### PR TITLE
Add fake setup.py

### PR DESCRIPTION
**Pull request by Kludex**
This is quite sad.

@michaeloliverx brought this to our attention on: https://github.com/encode/httpx/commit/45b7cfaad3a8987ea35fa5bf092bbdda485444fd#commitcomment-82042745.

The `Used by` section on the project's profile disappeared after we removed the `setup.py`. We can either accept this PR or revert https://github.com/encode/httpx/pull/2334, to have it back,

I'll let you folks decide @florimondmanca @tomchristie 🙏 

I'll update the Starlette and Uvicorn according to what is decided here.

Closes #2352

### CODE REVIEW

**Comment by michaeloliverx** [comment]
This is the same solution as `urllib3` has taken. You can read a bit more about it at https://sethmlarson.dev/blog/preparing-for-the-wave-of-open-source-funding#dependency-chain-metadata.

**Comment by Kludex** [comment]
Flask does this: https://github.com/pallets/flask/blob/main/setup.py#L3-L16

**Comment by florimondmanca** [comment]
That’s a raging GitHub limitation. I’d be okay with this PR; happier to use newest standards than stick to oldest because GH is lagging behind. The sad part is we’d have to do this and duplicate dependency declaration every project. So I’m on the fence…

**Comment by Kludex** [comment]
How we decide here?

**Comment by Kludex** [comment]
I've created #2352 as an alternative to this PR.

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import sys
+
+from setuptools import setup
```
Comment (samuelcolvin) [inline setup.py:3]: I guess technically this import isn't required since it can never be used.

Comment (Kludex) [inline setup.py:3]:
```suggestion
```

**Comment by Kludex** [comment]
> "Flask does this" is a pretty good argument. 😅
> 
> > The sad part is we’d have to do this and duplicate dependency declaration every project. So I’m on the fence…
> 
> Indeed. I'm not wild about it either, but the pallets team have this on all their repos. Probably an okay solution for now.

Since that's the case. Shall we choose this PR?

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import sys
```
Comment (Kludex) [inline setup.py:1]:
```suggestion
import sys

from setuptools import setup
```

**Review by michaeloliverx — APPROVED** [review]

**Comment by Kludex** [comment]
I'll merge this tonight if no objections are made. 🙏

**Pull request merged**